src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2261,17 +2312,129 @@ export class StudyViewPageStore {
         }, [] as ChartMeta[]);
     }
 
+    @computed
+    get showSettingRestoreMsg() {
+        if (this.isRestorePossible && this.fetchUserSettings.isComplete) {
+            return !_.isEqual(this.currentChartSettings, _.keyBy(this.fetchUserSettings.result, x => x.id));
+        }
+        return false;
+    }
+
     @computed
     get loadingInitialDataForSummaryTab() {
-        if (this.defaultVisibleAttributes.isPending ||
+        let pending = this.defaultVisibleAttributes.isPending ||
             this.initialVisibleAttributesClinicalDataBinCountData.isPending ||
             this.initialVisibleAttributesClinicalDataCountData.isPending ||
             this.mutationProfiles.isPending ||
-            this.cnaProfiles.isPending
-        ) {
-            return true;
-        } else {
-            return false;
+            this.cnaProfiles.isPending;
+
+        if (this.loadInitialUserSettings) {
+            pending = pending || this.fetchUserSettings.isPending
```

Comment:
When a user logged in while in study view, I think this will trigger a whole page refresh?